feat: deploy script for mainnet rewards

8sunyuan opened the pull request:
Main thing to look at is the configuration constants in `script/configs/mainnet/v0.3.0-mainnet-rewards.config.json`

ChaoticWalrus commented on script/deploy/mainnet/v0.3.0-mainnet-rewards.s.sol:
```diff
@@ -0,0 +1,100 @@
+        // STOP RECORDING TRANSACTIONS FOR DEPLOYMENT
+        vm.stopBroadcast();
+
+        // Sanity Checks
+        _verifyContractPointers();
+        _verifyImplementations();
+        _verifyContractsInitialized({isInitialDeployment: true});
+        _verifyInitializationParams();
```
so these checks aren't actually relevant to the new deployment?
would recommend replacing these with something more relevant, or at least adding some RewardsCoordinator-specific automated checks

8sunyuan replied:
The rewardsCoordinator params in the json `script/configs/mainnet/v0.3.0-mainnet-rewards.config.json` are checked in the ExistingDeploymentParser helper. Specifically the `_verifyInitializationParams` function call right here https://github.com/Layr-Labs/eigenlayer-contracts/blob/rewards-mainnet-deploy/script/utils/ExistingDeploymentParser.sol#L554-L593
Personally would like to keep the other functions in as additional sanity checks to ensure the correct addresses used (since we have different config files for holesky and mainnet)

ChaoticWalrus replied:
oh, neat! thanks, that's my bad; resolving

ChaoticWalrus commented on script/configs/mainnet/v0.3.0-mainnet-rewards.config.json:
```diff
@@ -0,0 +1,57 @@
+    "pauserMultisig": "0x5050389572f2d220ad927CcbeA0D406831012390",
+    "timelock": "0xA6Db1A8C5a981d1536266D2a393c5F8dDb210EAF"
+  },
+  "strategies": {
+    "numStrategies": 0,
+    "MAX_PER_DEPOSIT": 115792089237316195423570985008687907853269984665640564039457584007913129639935,
+    "MAX_TOTAL_DEPOSITS": 115792089237316195423570985008687907853269984665640564039457584007913129639935,
+    "strategiesToDeploy": []
+  },
+  "strategyManager": {
+    "init_strategy_whitelister": "0xBE1685C81aA44FF9FB319dD389addd9374383e90",
+    "init_paused_status": 0
```
would probably be good if we can clean these up by deleting irrelevant (to this deployment) parts of the config. I understand it not being worth it if deleting entries breaks other things though!

ChaoticWalrus reviewed (approved):
Looks reasonable to me.
I made a couple suggestions and there is some ongoing discussion around deployment details.
I'm not finding anything really amiss here though.

8sunyuan commented:
> Looks good to me. Can you deploy so we can go thru the checklist so we get the addresses in before merge?

Added deployment output file and added to README

ChaoticWalrus commented on README.md:
```diff
@@ -117,6 +117,7 @@ The current mainnet deployment is our M2 release. You can view the deployed cont
 | [`EigenPodManager`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/4b15d68b7e16b5965bad398496bfce57f5a47e1b/src/contracts/pods/EigenPodManager.sol) | [`0x91E677b07F7AF907ec9a428aafA9fc14a0d3A338`](https://etherscan.io/address/0x91E677b07F7AF907ec9a428aafA9fc14a0d3A338) | [`0xe429...5762`](https://etherscan.io/address/0xe4297e3dadbc7d99e26a2954820f514cb50c5762) | Proxy: [`TUP@4.7.1`](htt ...
 | [`AVSDirectory`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/4b15d68b7e16b5965bad398496bfce57f5a47e1b/src/contracts/core/AVSDirectory.sol) | [`0x135dda560e946695d6f155dacafc6f1f25c1f5af`](https://etherscan.io/address/0x135dda560e946695d6f155dacafc6f1f25c1f5af) | [`0xdabd...a5b7`](https://etherscan.io/address/0xdabdb3cd346b7d5f5779b0b614ede1cc9dcba5b7) | Proxy: [`TUP@4.7.1`](https://g ...
 | [`Slasher`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/4b15d68b7e16b5965bad398496bfce57f5a47e1b/src/contracts/core/Slasher.sol) | [`0xD92145c07f8Ed1D392c1B88017934E301CC1c3Cd`](https://etherscan.io/address/0xD92145c07f8Ed1D392c1B88017934E301CC1c3Cd) | [`0xf323...6614`](https://etherscan.io/address/0xf3234220163a757edf1e11a8a085638d9b236614) | Proxy: [`TUP@4.7.1`](https://github.com/ ...
+| [`RewardsCoordinator`](https://github.com/Layr-Labs/eigenlayer-contracts/blob/mainnet/src/contracts/core/RewardsCoordinator.sol) | [`0x7750d328b314EfFa365A0402CcfD489B80B0adda`](https://holesky.etherscan.io/address/0x7750d328b314EfFa365A0402CcfD489B80B0adda) | [`0x5bf7...8785`](https://etherscan.io/address/0x5bf7c13D5FAdba224ECB3D5C0a67A231D1628785) | Proxy: [`TUP@4.7.1`](https://github.com/Ope ...
```
the link for the Proxy is using Holesky Etherscan rather than mainnet Ehterscan -- the address is correct for mainnet though!

8sunyuan replied:
Thank you for this, fixed now https://github.com/Layr-Labs/eigenlayer-contracts/pull/638/commits/273c6ba2154906943ad8601596a644341d0ab137

simobuckz commented:
> Main thing to look at is the configuration constants in `script/configs/mainnet/v0.3.0-mainnet-rewards.config.json`

Hey is this the right contract address to invest in EIGEN.. 0x6aa666DE6102069555713d18E18Ef4404B87A79F .. Some guy on telegram is telling people that this is a good project and i believe it is, but i'm not able to verify this contract address. On Etherscan it does show there is a EIGEN account connected to that address but yeah. If you can please let me know i would greatly appreciate it.
